### PR TITLE
Embedded Tests

### LINKED ISSUE #45: Embedding by ID

### PR DESCRIPTION
**Pull request by epaulet**
fixes #45

### CODE REVIEW

**Comment by prayogoa** [comment]
:heart:

**Comment by epaulet** [comment]
Travis CI is failing for some reason, but it's not related to the test suite.

File: Gemfile [hunk 1]
```diff
@@ -3,6 +3,9 @@ source 'https://rubygems.org'
 # Specify your gem's dependencies in rainforest-cli.gemspec
 gemspec
 
+# Until API bits is updated
+gem 'rainforest', github: 'epaulet/rainforest-ruby', branch: 'rf-test-rfml-id'
```
Comment (ukd1) [inline Gemfile]: can you merge this code in to our main repo please?

File: lib/rainforest/cli/options.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ class OptionParser
     attr_writer :file_name, :tags
     attr_reader :command, :token, :tags, :conflict, :browsers, :site_id, :environment_id,
                 :import_file_name, :import_name, :custom_url, :description, :folder,
-                :debug, :file_name, :test_spec_folder
+                :debug, :file_name, :test_folder
```
Comment (ukd1) [inline lib/rainforest/cli/options.rb]: nice rename! :dancer:

File: Gemfile (hunk 1 above)
Comment (epaulet) [inline Gemfile]: I am not allowed to merge but I made a PR. https://github.com/rainforestapp/rainforest-ruby/pull/2

File: lib/rainforest/cli/test_importer.rb [hunk 3]
```diff
@@ -4,31 +4,29 @@
...
-#   b) second line must have a ?
-#   c) second line must not be blank
-# 2. comments:
-#   a) lines starting # are comments
+# Lines starting with # are test attributes or comments
+# Possible attributes: #{RainforestCli::TestParser::Parser::TEXT_FIELDS.join(', ')}
+#
+# Steps are composed of two lines: an action and a question. Example:
+#
+# This is a step action.
```
Comment (ukd1) [inline lib/rainforest/cli/test_importer.rb]: ``is the``

File: lib/rainforest/cli/test_importer.rb [hunk 4]
```diff
@@ -224,15 +186,117 @@ def validate
...
+    end
+    @_id_mappings
+  end
+
+  def create_test_obj(rfml_test)
+    test_obj = {
+      start_uri: rfml_test.start_uri || "/",
+      title: rfml_test.title,
+      description: rfml_test.description,
+      tags: (["ro"] + rfml_test.tags).uniq,
```
Comment (ukd1) [inline lib/rainforest/cli/test_importer.rb]: can we loose the ``ro`` tag yet?

File: lib/rainforest/cli/test_importer.rb [hunk 5]
```diff
@@ -224,15 +186,117 @@ def validate
...
+            redirection: true,
+            element: {
+              action: step.action,
+              response: step.response
+            }
+          }
+        when :test
+          {
+            type: 'test',
+            redirection: true,
```
Comment (ukd1) [inline lib/rainforest/cli/test_importer.rb]: this'll be the next thing to figure out

File: lib/rainforest/cli/test_parser.rb [hunk 6]
```diff
@@ -90,7 +101,7 @@ def process
         end
       end
 
-      if @test.id == nil
+      if @test.rfml_id == nil
         @test.errors[0] = Error.new(0, "Missing test ID. Please start a line #! followed by a unique id.")
```
Comment (ukd1) [inline lib/rainforest/cli/test_parser.rb]: "Missing RMFL ID"

File: lib/rainforest/cli/version.rb [hunk 7]
```diff
@@ -1,3 +1,3 @@
 module RainforestCli
-  VERSION = "1.1.4"
+  VERSION = "1.2.0"
```
Comment (ukd1) [inline lib/rainforest/cli/version.rb]: Please also update the changelog

File: lib/rainforest/cli/test_importer.rb (hunk 4 above)
Comment (epaulet) [inline lib/rainforest/cli/test_importer.rb]: Not yet. I'm just trying to get this working for now for some clients who are waiting on this to create their tests. That'll mostly be a server/frontend change I think.

**Pull request merged**